src/App.svelte

```diff
@@ -28,8 +28,11 @@
 
   const activeRouteStore = router.activeRouteStore;
 
-  void router.loadFromLocation();
-  httpd.initialize();
+  // This allows us to get the correct values for the httpd state and port, before navigation.
+  httpd
+    .initialize()
+    .then(() => void router.loadFromLocation())
+    .catch(() => void router.loadFromLocation());
```

Comment:
You can use
```suggestion
    .finally(() => router.loadFromLocation());
```

Reply:
hmm yeah that seems nicer, the thing is that eslint doesn't like the missing reject or catch clause
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-floating-promises.md

We could probably just ignore eslint on it in this case

Reply:
Addressed in f98c22b